twisted/internet/defer.py

```diff
@@ -661,6 +661,66 @@ def __str__(self):
     __repr__ = __str__
 
 
+    def __await__(self):
+        awaiter = _MakeDeferredAwaitable(self)
```

Comment:
I think for now I'll keep it like this -- it's getting a bit complex to review for those that don't know it as well as either of us do, so I can do it once I have a benchmark :)